tests/apicallcaptor.py

```diff
@@ -0,0 +1,180 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+import unittest
+from sys import stderr
+
+from botocore.history import get_global_history_recorder
+
+
+class ApiCallCaptor:
```

Comment:
This is a cool use of botocore's history recording :+1: . I do want to be a bit careful though, since tests have both a value and a cost, and I don't want to get too hung up on testing botocore's underlying activity. It seems very easy to end up with a glut of tests at a finer grained level of detail than we want to maintain for the value we get out of them.

Reply:
See previous comment re this being a test of my code and cc config rather than the APIs